packages/block-editor/src/components/block-settings-menu/index.js

```diff
@@ -112,6 +113,7 @@ export function BlockSettingsMenu( { clientIds } ) {
 								<MenuGroup>
 									{ ! isLocked && (
 										<MenuItem
+											disabled={ ! canRemove }
```

Comment:
in php, we might use a filter:

```php
$disabled = apply_filters( 'should_disable_block_remove_button', false, block_name );
```

I wonder what the best approach for doing something like that would be here. I'm familiar-ish with slot fill, but not familiar with JS filters/actions in Gutenberg.

Do we need to implement a store for something like how we remove settings panels?